rusk/src/lib/chain/rusk.rs

```diff
@@ -435,9 +443,16 @@ fn accept(
             .checked_sub(gas_spent)
             .ok_or(Error::OutOfGas)?;
 
+        let economic_gas_spent = match receipt.economic_mode {
+            Allowance(allowance) if allowance > 0 => 0,
+            Charge(charge) if charge > 0 => charge,
+            _ => gas_spent,
```

Comment:
I find this very hard to understand.

If allowance is greater then 0 then the economic gas spent is 0
I believe the real culprit is the following: "What is economic_gas_spent"?
Is the amount spent by the contract? Is the amount charged to the user?

Furthermore, sometime the economic_gas_spent is just GasPoints (`gas_spent`), and sometimes is the whole fee paid (`charge`)

Reply:
economic_gas_spent is always in gas points, never in value, I corrected some code to make sure this is the case, I hope it will be clear now